File: Algorithm/Python/929. Unique Email Addresses.py
# -*- coding: utf-8 -*-
# Created on Sun Oct 28 2018 17:4:17
# Author: WuLC
# EMail: liangchaowu5@gmail.com

# simple solution
class Solution(object):
    def numUniqueEmails(self, emails):
        """
        :type emails: List[str]
        :rtype: int
        """
        def format(s):
            result = []
            for c in s:
                if c == '+':
                    break
                elif c == '.':
                    continue
                else:
                    result.append(c)
            return ''.join(result)
        uniuqe_emails = set()
        for email in emails:
            name, domain = email.split('@')
            uniuqe_emails.add(format(name) + '@' + domain)
        return len(uniuqe_emails)

File: Algorithm/C++/929. Unique Email Addresses.cpp
/*
 * Created on Sun Oct 28 2018 17:8:57
 * Author: WuLC
 * EMail: liangchaowu5@gmail.com
 */

// simple solution
class Solution {
 public:
    int numUniqueEmails(vector<string>& emails) {
      set<string> result;
      for(auto email : emails) {
        string tmp = "";
        int idx = 0;
        bool abandon = false;
        while (email[idx] != '@'){
          if (email[idx] == '+')
            abandon = true;
          else if (abandon == false && email[idx] != '.')
            tmp += email[idx];
          idx += 1;
        }
        tmp += email.substr(idx);
        result.insert(tmp);
      }
      return result.size();
    }
};